### PR TITLE
Fix indentation on named arguments of functions

### LINKED ISSUE #335: Missing indentation on named arguments of functions
When formatting function calls with 'named arguments' (not sure what they are called) the formatter does not indent the arguments in the same way as for unnamed arguments.

Example:
```systemverilog
  always_comb begin
    value = function_name(
    .long_parameter(8'hA),
    .another_parameter(signal_123),
    .parameter_three(foobar)
    );
  end
```

For unnamed arguments (if the line were long enough) this would be:
```systemverilog
  always_comb begin
    value = function_name(8'hA, signal_123,
                          foobar);
  end
```

Adding some indentation before the arguments seems more readable to me:
```systemverilog
  always_comb begin
    value = function_name(
      .long_parameter(8'hA),
      .another_parameter(signal_123),
      .parameter_three(foobar)
    );
  end
```

### PR DESCRIPTION
**Pull request by wtatarski**
Fixes #335

### CODE REVIEW

**Review by tgorochowik — COMMENTED** [review]
LGTM in general.

But I think we should add a test that mixes named and positional arguments (note that positional arguments must come fist, see 13.5.4 in LRM)

**Comment by wtatarski** [comment]
Added test with mixed named and positional arguments.

**Review by tgorochowik — APPROVED** [review]

**Review by hzeller — COMMENTED** [review]
Generally looks good though I see that there might be some corner cases, which you should ask about and discuss on #335 so that some decisions of how to behave in each of these cases can be made there and referred to later in case there are questions. Pull in some of the OpenTitan folks who can give good guidance (See #445 for some such discussion about corner cases).

Otherwise we might flip-flop between competing implementations later, which is probably better to clear out first.

**Comment by wtatarski** [comment]
@hzeller 
I think that all cases which you mentioned are fixed now.

**Review by hzeller — APPROVED** [review]
LGTM

**Comment by hzeller** [comment]
Please test this with OpenTitan formatting and gather potential feedback from that team in #335 so that we can potentially refine.

File: verilog/formatting/formatter.cc [hunk 1]
```diff
@@ -379,6 +396,45 @@ static void DeterminePartitionExpansion(
         node_view.Expand();
       }
     }
+
```
Comment (hzeller) [inline verilog/formatting/formatter.cc:399]: Is there a break missing here @wtatarski (or: ABSL_FALLTHROUGH_INTENDED) ?
(also a test should trigger if the uintended situation is happening)

Please add a follow-up CL to clarify this part of the code (and possibly add a test that makes sure that this is caught)
